File: SkillDipz/frontend/src/types/targetCompany.ts
export type EligibilityStatus =
  | "eligible"
  | "not_yet"
  | "skill_gap"
  | "full_match";

export interface InterviewRound {
  order: number;
  name: string;
  description?: string;
  duration_mins?: number;
}

export interface MatchedCompany {
  company_id: string;
  name: string;
  logo_emoji?: string;
  logo_url?: string;
  industry: string;
  website?: string;
  headquarters?: string;
  min_score: number;
  your_score: number;
  eligible: boolean;
  eligibility_status: EligibilityStatus;
  skill_match_pct: number;
  score_readiness_pct: number;
  match_score: number;
  matched_skills: string[];
  missing_skills: string[];
  interview_rounds: InterviewRound[];
  active_openings: number;
  match_rank: number;
  selected_by: "student" | "auto_suggested";
  last_recomputed_at?: string;
}

export interface NotYetEligibleCompany {
  company_id: string;
  name: string;
  logo_emoji?: string;
  logo_url?: string;
  industry: string;
  min_score: number;
  your_score: number;
  score_gap: number;
  missing_skills: string[];
  active_openings: number;
}

export interface TargetCompaniesResponse {
  student_score: number;
  student_role: string;
  selected_companies: MatchedCompany[];
  auto_suggested: MatchedCompany[];
  companies_not_yet_eligible: NotYetEligibleCompany[];
  last_updated_at?: string;
}

export interface SelectCompanyResponse {
  message: string;
  company_id: string;
  match_result: MatchedCompany;
}

export interface CompanyProfileDetail {
  company_id: string;
  name: string;
  logo_emoji?: string;
  logo_url?: string;
  industry: string;
  website?: string;
  headquarters?: string;
  description?: string;
  required_roles: string[];
  must_have_skills: string[];
  nice_to_have_skills: string[];
  min_score: number;
  interview_rounds: InterviewRound[];
  interview_tips?: string;
  active_openings: number;
  is_verified: boolean;
}

export interface VerifiedCompany {
  company_id: string;
  name: string;
  logo_emoji?: string;
  logo_url?: string;
  industry: string;
  headquarters?: string;
  required_roles: string[];
  must_have_skills: string[];
  min_score: number;
  active_openings: number;
}

File: SkillDipz/backend/app/models/target_company.py
from sys import flags
from beanie import Document
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime, timezone
from enum import Enum


class EligibalityStatus(str, Enum):
    ELIGIBLE = "eligible"
    NOT_YET = "not_yet"
    SKILL_GAP = "skill_gap"
    FULL_MATCH = "full_match"


class InterviewRound(BaseModel):
    order: int
    name: str
    description: Optional[str] = None
    duration_mins: Optional[int] = None


class CompanyProfile(Document):
    comapny_id: str
    name: str
    logo_emoji: Optional[str] = None
    logo_url: Optional[str] = None
    industry: str
    website: Optional[str] = None
    description: Optional[str] = None
    headquaters: Optional[str] = None

    # Matching criteria (set when company posts a job)
    required_roles: List[str] = []
    must_have_skills: List[str] = []
    nice_to_have_skills: List[str] = []
    min_score: float = 0.0

    # Interview Process
    interview_rounds: List[InterviewRound] = []
    interview_tips: Optional[str] = None

    # Platform state
    is_verified: bool = False
    active_openings_count: int = 0

    registered_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))

    class Settings:
        name = "company_profiles"

class StudentTargetCompany(Document):
    student_id: str
    company_id: str
    
    # Selection metadata
    selected_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    selected_by: str = "student"


    # Match result (recomputed on trigger)
    match_score: float = 0.0
    skill_match_pct: float = 0.0
    score_readiness_pct: float = 0.0
    eligibality_status: EligibalityStatus = EligibalityStatus.NOT_YET

    matched_skills: List[str] = []
    missing_skills: List[str] = []
    match_rank: int = 0


    last_recomputed_at: Optional[datetime] = None
    notification_sent: bool = False

    class Settings:
        name = "student_target_companies"